From 2adc992d2657225863f570bc87bba0486d4d18a3 Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Thu, 24 Jan 2019 18:43:37 +0800
Subject: [PATCH 2/2] drivers: spi: set auto_runtime_pm as true during driver
 probe

In upstream commit 9e3a000362ae ("spi: zynqmp: Add pm
runtime support"), implement qspi power management. But compared
with sdk commit 3bae015728a6("spi: zynqmp: Add pm runtime support"),
the mainline code omit to set auto_runtime_pm as true.
Therefore add code to set auto_runtime_pm as true so that qspi clock is
enabled before operating qspi device.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/spi/spi-zynqmp-gqspi.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/spi/spi-zynqmp-gqspi.c b/drivers/spi/spi-zynqmp-gqspi.c
index 5ee3aeb..915b444 100644
--- a/drivers/spi/spi-zynqmp-gqspi.c
+++ b/drivers/spi/spi-zynqmp-gqspi.c
@@ -1315,6 +1315,7 @@ static int zynqmp_qspi_probe(struct platform_device *pdev)
 	master->mode_bits = SPI_CPOL | SPI_CPHA | SPI_RX_DUAL | SPI_RX_QUAD |
 			    SPI_TX_DUAL | SPI_TX_QUAD;
 	xqspi->speed_hz = master->max_speed_hz;
+	master->auto_runtime_pm = true;
 
 	if (master->dev.parent == NULL)
 		master->dev.parent = &master->dev;
-- 
1.7.5.4

